packages/core/src/common/promise-util.ts

```diff
@@ -17,4 +17,16 @@ export class Deferred<T> {
         this.resolve = resolve;
         this.reject = reject;
     });
+
+    static resolved<T>(resolution: T): Deferred<T> {
+        const deferred = new Deferred<T>();
+        deferred.resolve(resolution);
```

Comment:
I don't think this is generally useful, as it somewhat counters the purpose of Deferred. 
Could you just inline the two lines and remove these methods again?